packages/upstream/x509.0.8.1+1/files/0001-signature-algorithm.patch

```diff
@@ -0,0 +1,26 @@
+diff --git a/lib/certificate.ml b/lib/certificate.ml
+index 6297dd3..5878d73 100644
+--- a/lib/certificate.ml
++++ b/lib/certificate.ml
+@@ -196,6 +196,8 @@ let validity { asn ; _ } = asn.tbs_cert.validity
+ 
+ let public_key { asn = cert ; _ } = cert.tbs_cert.pk_info
+ 
++let signature_algorithm { asn = cert ; _ } = Algorithm.to_signature_algorithm cert.signature_algo
++
+ let supports_keytype c t =
+   match public_key c, t with
+   | (`RSA _), `RSA -> true
+diff --git a/lib/x509.mli b/lib/x509.mli
+index d94f924..30bc0cb 100644
+--- a/lib/x509.mli
++++ b/lib/x509.mli
+@@ -427,6 +427,8 @@ module Certificate : sig
+ 
+   (** [extensions certificate] is the extension map of [certificate]. *)
+   val extensions : t -> Extension.t
++
++  val signature_algorithm : t -> ([> `ECDSA | `RSA ] * [> `MD5 | `SHA1 | `SHA224 | `SHA256 | `SHA384 | `SHA512 ]) option
```

Comment:
This should perhaps have a type defined, but since this is a just temporary change until upstream updates a smaller change like here is probably better.